packages/versioning/src/versioning.ts

```diff
@@ -343,6 +343,13 @@ export class VersionMap {
 
 export function $versioned(context: DecoratorContext, t: Namespace, versions: Enum) {
   context.program.stateMap(versionsKey).set(t, new VersionMap(t, versions));
+  // associate a copy of the map with the enum's containing namespace as well
+  const enumNamespace = versions.namespace;
+  if (enumNamespace) {
+    context.program
+      .stateMap(versionsKey)
+      .set(enumNamespace, new VersionMap(enumNamespace, versions));
+  }
```

Comment:
This was the easy fix. It feels VERY hacky to me.

Reply:
So, if the enum is in the global namespace then this adds the version to the global namespace?

Reply:
Yeah, agreed, so, do we need the reverse lookup table from enuto the namespace where it is applied at all?  I don't find any instances where we use the namespace from getVersionForEnum.

And since dependencies are now applied to the version anyway, feels like we can maybe do away with this

Reply:
I'm reluctant to mess with the existing logic because I think the state map Namespace key is different from the VersionMap Namespace key, but what I did instead is create a reverse lookup table where you can pass in the enum and get whatever namespace(s) version based on it. This is a lot less hacky than what I tried before and works for our scenarios.